sale_order_import/wizard/import_sale.py

```diff
@@ -223,167 +520,190 @@ def import_sale_data(self):
                 quotechar='"',
                 delimiter=','
             )
-            fields = next(csv_iterator)
-
-            order_group = fields.index('Group')
-            partner_id = fields.index('Customer')
-            product_id = fields.index('Line Product')
-            line_name = fields.index('Line Description')
-            price_unit = fields.index('Line Unit Price')
-            product_qty = fields.index('Line Qty')
-            taxes_id = fields.index('Line Tax')
-            notes = fields.index('Notes')
-            pricelist_id = fields.index('Pricelist')
-            warehouse_id = fields.index('Warehouse')
+            sheet_fields = next(csv_iterator)
+
+            #  column validation
+            missing_columns = list(set(FIELDS_TO_IMPORT) - set(sheet_fields))
+            if missing_columns:
+                raise Warning(
+                    _('Following columns are missing: \n %s'
+                      % ('\n'.join(missing_columns))
+                      )
+                )
+
```

Comment:
Below lines are too redundant.  Replace them by:
```
        order_group = sheet_fields.index('Group')
        partner_name = sheet_fields.index('Customer')
        partner_tel = sheet_fields.index('Customer Phone/Mobile')
        product_id = sheet_fields.index('Line Product')
        line_name = sheet_fields.index('Line Description')
        price_unit = sheet_fields.index('Line Unit Price')
        product_qty = sheet_fields.index('Line Qty')
        taxes_id = sheet_fields.index('Line Tax')
        notes = sheet_fields.index('Notes')
        pricelist_id = sheet_fields.index('Pricelist')
        warehouse_id = sheet_fields.index('Warehouse')
        team_id = sheet_fields.index('Team')
        carrier_id = sheet_fields.index('Carrier')
```

Or there may be a smarter way to do this.